14.0 add shopinvader customer delivery

mourad-ehm opened the pull request.

mourad-ehm committed:
[ADD][14.0] module shopinvader_customer_delivery

mourad-ehm committed:
[FIX][14.0] manifest

mourad-ehm commented:
@hparfr

hparfr commented on shopinvader_customer_delivery/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+    "category": "shopinvader",
+    "version": "14.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["account", "shopinvader"],
```
account?

hparfr replied:
still this one

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,203 @@
+class DeliveryMoveService(Component):
+    """Service for getting information on stock.picking"""
+
+    _inherit = ["base.shopinvader.service", "abstract.shopinvader.download"]
```
why download ?

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,203 @@
+        sale_domain = sale_service._get_base_search_domain()
+        print(sale_domain)
+        sales = self.env[sale_service._expose_model].search(sale_domain)
+        print([("sale_id", "in", sales.ids)] + self._get_allowed_stock_move_domain())
```
print ?!

mourad-ehm committed:
[FIX][14.0] shopinvader_customer_delivery: code review

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,201 @@
+        domain = [
+            ("picking_id.picking_type_id.code", "=", "outgoing"),
+        ]
+        if states:
```
useless condition

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,201 @@
+        ]
+        return to_parse
+
+    def _add_product_info(self, stok_move):
```
typo

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,201 @@
+        :param picking: stock.move
+        :return: dict
+        """
+        sale_order = stok_move.product_id
```
wrong  variable name

hparfr replied:
@mourad-ehm please fix !

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,201 @@
+        values = sale_order.jsonify(parser, one=True)
+        return values
+
+    def _get_parser_sale_order(self):
```
wrong variable name

mourad-ehm replied:
Which ?

mourad-ehm committed:
[FIX][14.0] shopinvader_customer_delivery: fix search by seller

mourad-ehm committed:
[FIX][14.0] shopinvader_customer_delivery: pre-commit

mourad-ehm committed:
[FIX][14.0] shopinvader_customer_delivery: get coming deliveries for connected user

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,226 @@
+        ]
+        return to_parse
+
+    def _add_product_info(self, stok_move):
```
```suggestion
    def _add_product_info(self, stock_move):
```

hparfr commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,226 @@
+        :return: dict
+        """
+        sale_order = stok_move.product_id
+        if not sale_order:
+            return {}
+        parser = self._get_parser_sale_order()
+        values = sale_order.jsonify(parser, one=True)
+        return values
```
```suggestion
        if not product:
            return {}
        parser = self._get_parser_product()
        values = product.jsonify(parser, one=True)
        return values
```

mourad-ehm commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,226 @@
+        :param picking: stock.move
+        :return: dict
+        """
+        sale_order = stok_move.product_id
```
```suggestion
        sale_order = stock_move.product_id
```

mourad-ehm committed:
Apply suggestions from code review

Typo & product parser

Co-authored-by: Hpar <raph@hpar.fr>

mourad-ehm commented on shopinvader_customer_delivery/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+    "category": "shopinvader",
+    "version": "14.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["account", "shopinvader"],
```
```suggestion
    "depends": ["stock", "shopinvader"],
```

mourad-ehm replied:
```suggestion
    "depends": ["stock", "shopinvader"],
```

mourad-ehm commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,226 @@
+            ("picking_id.picking_type_id.code", "=", "outgoing"),
+        ]
+        if states:
+            domain.append(("state", "in", states))
```
```suggestion
```

mourad-ehm commented on shopinvader_customer_delivery/services/stock_move.py:
```diff
@@ -0,0 +1,226 @@
+        domain = [
+            ("picking_id.picking_type_id.code", "=", "outgoing"),
+        ]
+        if states:
```
```suggestion
        domain.append(("state", "in", states))
```

mourad-ehm committed:
Apply suggestions from code review

Dependency, useless condition, typo

mourad-ehm committed:
Apply suggestions from code review

useless condition

mourad-ehm committed:
[FIX][14.0] shopinvader_customer_delivery: product parser

paradoxxxzero committed:
[IMP] shopinvader_customer_delivery: Handle seller authentication

Add shopinvader_seller_access as dependency
Change filter behavior to uniformize with other seller access services
Add unit tests
Fix typos
